lib/ClangImporter/ClangImporter.cpp

```diff
@@ -779,6 +779,7 @@ bool ClangImporter::canReadPCH(StringRef PCHFilename) {
   invocation->getPreprocessorOpts().DisablePCHValidation = false;
   invocation->getPreprocessorOpts().AllowPCHWithCompilerErrors = false;
   invocation->getHeaderSearchOpts().ModulesValidateSystemHeaders = true;
+  invocation->getHeaderSearchOpts().ValidateASTInputFilesContent = true;
   invocation->getLangOpts()->NeededByPCHOrCompilationUsesPCH = true;
```

Comment:
In theory this shouldn't be necessary, since it should get picked from the clone of the main instance's invocation above.  Same with `NeededByPCHOrCompilationUsesPCH`.  (The others aren't explicitly set in the main instance, so it's not obvious they can be skipped here.)

Reply:
I'm not sure about this: when I added `NeededByPCHOrCompilationUsesPCH` I remember I needed to add extra ones otherwise some testcases would fail, maybe at the time we had lifetime issues with these instances? Lemme take a look again